### PR TITLE
Remove ops attribute from Optimizer

### PR DESCRIPTION
**Pull request by honnibal**
The `Optimizer` object can't assume that all layers are using the same backend, because we might have mixed CPU/GPU execution. So the optimizer needs to fetch an ops appropriate to the inputs.

We remove the `ops` argument to the optimizer, and add a helper `get_array_ops`.

### CODE REVIEW

File: thinc/model.py [hunk 1]
```diff
@@ -304,12 +306,12 @@ def finish_update(self, optimizer: Optimizer) -> None:
                 if node.has_grad(name):
                     param = node.get_param(name)
                     grad = node.get_grad(name)
-                    if hasattr(optimizer, 'ops'):
-                        param = optimizer.ops.asarray(param)        # type: ignore
-                        grad = optimizer.ops.asarray(grad)          # type: ignore
+                    if hasattr(optimizer, "ops"):
```
Comment (svlandeg) [inline thinc/model.py:309]: is this check still useful ?

Comment (honnibal) [reply thinc/model.py:309]: It's not, no. We should remove that code

Comment (honnibal) [inline thinc/model.py:309]:
```suggestion

```

File: thinc/model.py [hunk 2]
```diff
@@ -304,12 +306,12 @@ def finish_update(self, optimizer: Optimizer) -> None:
...
+                        param = optimizer.ops.asarray(param)  # type: ignore
```
Comment (honnibal) [inline thinc/model.py:310]:
```suggestion
```

File: thinc/model.py [hunk 3]
```diff
@@ -304,12 +306,12 @@ def finish_update(self, optimizer: Optimizer) -> None:
...
+                        grad = optimizer.ops.asarray(grad)  # type: ignore
```
Comment (honnibal) [inline thinc/model.py:311]:
```suggestion
```

**Pull request merged**